src/popup/utils/utils.ts

```diff
@@ -727,3 +727,9 @@ export function connectFrames(sdk: ISdk | AeSdkWallet) {
     handleUnknownError(error);
   }
 }
+
+export function checkImageAvailability(url: string): Promise<boolean> {
+  return fetch(url)
```

Comment:
Instead of fetching whole file I suggest to only fetch headers: https://stackoverflow.com/questions/47240975/how-to-fetch-using-fetch-api-only-headers-in-javascript-not-the-whole-content